### PR TITLE
jormun: migration to python3

### PR DESCRIPTION
**Pull request by kinnou02**
This start the migration process to python 3

This PR doesn't (yet?) address the fact that protoc doesn't generate python code that works with python3...

There is a few hack that need to be improved before merging.
Each commit should be understandable. 
Call to `cmp_to_key` should be replaced by an implementation that works natively with `key`. 

At this point all integration tests are OK, jormun unittest remains KO :(

TODO:
  - [ ] understand (and improve) https://github.com/CanalTP/navitia/pull/2989/commits/65e7e4f40949d41bb1dd4b77edd1694795d20a0c
  - [x] make protobuf file ok (probably with `2to3`)
  - [ ] CI

Obviously this still works with python 2

### CODE REVIEW

File: source/jormungandr/tests/graphical_isochrones_tests.py [hunk 1]
```diff
@@ -351,7 +351,7 @@ def test_graphical_isochrones_invalid_datetime(self):
         normal_response, error_code = self.query_no_assert(q)
 
         assert error_code == 400
-        assert 'Unable to parse datetime, year is out of range' in normal_response['message']
+        assert 'Unable to parse datetime, year' in normal_response['message']
```
Comment (benoit-bst) [inline source/jormungandr/tests/graphical_isochrones_tests.py:354]: Dunno why the message has changed ? :eyes: 
The python compatibility should not change that ?

File: source/jormungandr/tests/heat_maps_tests.py [hunk 2]
```diff
@@ -228,7 +228,7 @@ def test_heat_maps_invalid_datetime(self):
         normal_response, error_code = self.query_no_assert(q)
 
         assert error_code == 400
-        assert 'Unable to parse datetime, year is out of range' in normal_response['message']
+        assert 'Unable to parse datetime, year' in normal_response['message']
```
Comment (benoit-bst) [inline source/jormungandr/tests/heat_maps_tests.py:231]: same

File: source/jormungandr/tests/graphical_isochrones_tests.py (hunk 1 above)
Comment (kinnou02) [reply source/jormungandr/tests/graphical_isochrones_tests.py:354]: This is the exception message from the datetime parser, in python 3 they improved the message to provide the value that was "parsed"

File: source/type/CMakeLists.txt [hunk 3]
```diff
@@ -59,20 +59,23 @@ add_custom_command(OUTPUT ${PROTO_HDRS} ${PROTO_SRCS}
...
     type.proto response.proto request.proto task.proto stat.proto
+    COMMAND 2to3 -w ${NAVITIACOMMON_PROTO_PY}
     WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}/navitia-proto" VERBATIM
     DEPENDS ${PROTO_FILES})
 
 #we don't want a dependancy to navitiacommon for monitor
 add_custom_command (OUTPUT ${MONITOR_KRAKEN_PROTO_PY}
     COMMAND protoc ARGS "--python_out=${MONITOR_KRAKEN_DIR}"
     type.proto response.proto request.proto
+    COMMAND 2to3 -w ${MONITOR_KRAKEN_PROTO_PY}
```
Comment (nonifier) [inline source/type/CMakeLists.txt:70]: :+1:

**Review by nonifier — APPROVED** [review]

**Review by mbounabi — APPROVED** [review]

**Comment by kinnou02** [comment]
I will rebase when we agree to merge it

**Review by benoit-bst — APPROVED** [review]

**Comment by kinnou02** [comment]
Jenkins, test this please.

**Comment by kinnou02** [comment]
Jenkins, test this please.

**Pull request merged**